src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/AbstractItemNetwork.java

```diff
@@ -432,7 +460,7 @@ private void gatherItemsFromBarrel(Location l, BlockMenu blockMenu, List<ItemSta
             for (int slot : blockMenu.getPreset().getSlotsAccessedByItemTransport((DirtyChestMenu) blockMenu, ItemTransportFlow.WITHDRAW, null)) {
                 ItemStack is = blockMenu.getItemInSlot(slot);
 
-                if (is != null && CargoUtils.matchesFilter(l.getBlock(), is)) {
+                if (is != null && CargoUtils.matchesFilter(this, l.getBlock(), is)) {
```

Comment:
Im not a fan of the name is because when reading it reads like:
`if is is not null`

However this is just my opinion